Fix mismatch issue with SGEMM(and HGEMM,CGEMM) + DirectToLds + GlobalReadVectorWidth > 1 + TailLoop

nakajee opened the pull request:
* without this condition, pre_checkin test dtv_sgemm_lite.yaml fails

benjaminulmer commented:
Seems like this might be a case of whac-a-mole: the dtv_sgemm_lite.yaml passed on this commit, but now dtl_hgemm.yaml is failing. 

@nakajee @AlexBrownAMD let's discuss tomorrow (June 24th) morning, but if this PR just substitutes one failure for another I think we should leave it. We can always hot-fix if needed once we understand the issue better.

nakajee commented:
It looks like the fail is because my change makes that case 0 valid solution.
We can remove the case at this moment.

nakajee commented:
> It looks like the fail is because my change makes that case 0 valid solution. We can remove the case at this moment.

I just realized I already knew how to fix the original fail (means I already new the real fix)....
I already have a solution for dgemm. I just need to apply same fix for other types (remove type limitaion).
I confirmed dtv_sgemm_lite.yaml passed with that fix (no need to add the reject condition back).

nakajee commented:
I also confirmed the same issue happened with HGEMM and my change fixed the issue with HGEMM.
I updated the title of this PR (also added "+ TailLoop").

benjaminulmer reviewed (approved):
Problem pipeline passed with latest commit and it doesn't look like anything else broke as a result.
